### PR TITLE
Localstorage and history sync

### PR DESCRIPTION
**Pull request by nakxy**
Have to sync up on
-Playlist/ History API spec
-Integrate Playlist/History API
-Integrate user logedin localstorage

### CODE REVIEW

File: app/components/SideBar.js [hunk 1]
```diff
@@ -21,12 +21,10 @@ export default class SideBar extends Component {
...
     }
 
-    buildUserPlaylist(userPlaylist){
+    buildUserPlaylist(userList){
     	let fields = [];
-    	for(let item in userPlaylist){
-			if(item !== 'favourites' && item !== 'history'){
-    			fields.push(<li><Link to={"/userList/"+item} activeClassName="active">{item}</Link></li>)
-			}
+    	for(let item in userList){
```
Comment (shreyasg14) [inline app/components/SideBar.js:26]: ad fav check back. its still inside userlist. We'll be showing fav twice in this case.

File: app/routes.js [hunk 2]
```diff
@@ -16,6 +16,8 @@ export default (
 		<Route path="/nowPlaying(/:play)(/:list)" component={NowPlaying} />
 		<Route path="/newReleases(/:play)(/:list)" component={Playlist} />
 		<Route path="/mostPopular(/:play)(/:list)" component={Playlist} />
+		<Route path="/history(/:play)(/:list)" component={Playlist} />
+		<Route path="/favourite(/:play)(/:list)" component={Playlist} />
```
Comment (shreyasg14) [inline app/routes.js:20]: fav route not needed.

File: app/actions/home.js [hunk 3]
```diff
@@ -191,4 +202,17 @@ export function getRedditList(subReddit){
...
 	}
+}
+// Have to test this, just stubbing the function for now
+export function postPlaylist(userID, name, tracks){
+	return(dispatch, getState)=> {
+		let state = getState();
+		axios({
+		  method:'post',
+		  url:'/api/v1/reddit?subReddit='+subReddit,
+		  body:JSON.stringify({usserId:userID,playlistName:[name],tracks:tracks})
```
Comment (shreyasg14) [inline app/actions/home.js]: send the payload like I sent in fav api call. 
{
      playlists: {
             [name]: [tracks]
     }
}

Comment (shreyasg14) [reply app/actions/home.js]: I read the comment on top of the method now 💯

File: app/components/SideBar.js (hunk 1 above)
Comment (nakxy) [reply app/components/SideBar.js:26]: i will change it, forgot to make those changes

File: app/actions/home.js (hunk 3 above)
Comment (nakxy) [reply app/actions/home.js]: 👍

**Pull request merged**